src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Rss20FeedFormatter.cs

```diff
@@ -20,6 +21,7 @@ public class Rss20FeedFormatter : SyndicationFeedFormatter, IXmlSerializable
         private static readonly XmlQualifiedName s_rss20Url = new XmlQualifiedName(Rss20Constants.UrlTag, string.Empty);
         private const string Rfc822OutputLocalDateTimeFormat = "ddd, dd MMM yyyy HH:mm:ss zzz";
         private const string Rfc822OutputUtcDateTimeFormat = "ddd, dd MMM yyyy HH:mm:ss Z";
+        private static readonly Regex PersonDisplayRegex = new Regex(@"(.*@[^ ]+)[ ]*\((.*)\)", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
IMO this should use the new [`GeneratedRegex`](https://github.com/dotnet/runtime/issues/62123#issuecomment-1191789133) source generator instead.
(note: this is currently being renamed from `RegexGenerator` to `GeneratedRegex`, so take care if you're going to look for further infos on that)

Reply:
I thought about that, but was not sure since I'm seeing some work still being done (like the renaming you mentioned): I supposed the best way would've been to start pushing this and maybe change later on, when work on the GeneratedRegex attribute will be finished (I see it's not yet currently merged).

Let me know how you would like to go on with this, thanks!

Reply:
This library builds for downlevel targets, so it can't only use GeneratedRegex.  If this PR is desired by this library's owners, it should use GeneratedRegex on .NET 7 and Regex elsewhere, e.g.
```C#
private const string PersonDisplayRegexPattern = @"(.*@[^ ]+)[ ]*\((.*)\)";
#if NET_7_OR_GREATER
    [GeneratedRegex(PersonDisplayRegexPattern)]
    private static partial Regex PersonDisplayRegex();
#else
    private static Regex PersonDisplayRegex() => s_personDisplayRegex;
    private static readonly Regex s_personDisplayRegex = new Regex(PersonDisplayRegexPattern, RegexOptions.Compiled);
#endif
```
Note that RegexOptions.CultureInvariant is a nop in how you've used it, so I removed it from my example.

Reply:
FWIW I agree with the approach, makes total sense.
Maybe we can wait for the GeneratedRegex attribute rename to land on `main`, and then apply the suggested changes to support both scenarios.

Sounds good?

Reply:
Change done